File: backend/src/main/java/com/fdsa/altairis/service/HotelsService.java
package com.fdsa.altairis.service;

import com.fdsa.altairis.dto.request.HotelRequest;
import com.fdsa.altairis.dto.response.HotelResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface HotelsService {
    HotelResponse create(HotelRequest request, String user);
    HotelResponse update(Long id, HotelRequest request, String user);
    HotelResponse delete(Long id, String user);
    HotelResponse getById(Long id);
    Page<HotelResponse> search(String q, Long cityId, String status, Pageable pageable);
}

File: frontend/src/app/services/services/hotels.service.ts
import { Injectable } from '@angular/core';
import { ApiClient } from '../../core/api/api.client';
import { PageResponse } from '../../core/api/api.types';
import { HotelCreateRequest, HotelUpdateRequest } from '../../interfaces/request/hotel.request';
import { HotelResponse } from '../../interfaces/response/hotel.response';

@Injectable({ providedIn: 'root' })
export class HotelsService {
  constructor(private api: ApiClient) {}

  search(q: string | null, cityId: number | null, status: string | null, page: number, size: number) {
    return this.api.get<PageResponse<HotelResponse>>('/api/hotels', {
      q: q ?? undefined,
      cityId: cityId ?? undefined,
      status: status ?? undefined,
      page,
      size,
    });
  }

  getById(id: number) {
    return this.api.get<HotelResponse>(`/api/hotels/${id}`);
  }

  create(req: HotelCreateRequest) {
    return this.api.post<HotelResponse>('/api/hotels', req);
  }

  update(id: number, req: HotelUpdateRequest) {
    return this.api.put<HotelResponse>(`/api/hotels/${id}`, req);
  }

  delete(id: number) {
    return this.api.delete<HotelResponse>(`/api/hotels/${id}`);
  }
}
